build/azure-pipelines/linux/sql-product-build-linux.yml

```diff
@@ -129,16 +129,19 @@ steps:
   - script: |
       set -e
       yarn gulp vscode-linux-x64-build-deb
+      du -ah ../ | sort -n -r
```

Comment:
Instead of using relative paths I'd suggest just using the environment var $(agent.workfolder) since that'll include the entire working folder for the agent job.

Although I have a few concerns that I haven't been able to find answers to. 

1. Does the size limit include the entire VM? There's stuff installed by the agent outside the working folder too and if that's getting counted that might explain why we're seeing these failures without changes on our end
1. Is the Hosted agent VM being cleaned up after every job? My gut instinct would be "I hope it is" but https://developercommunity.visualstudio.com/content/problem/566745/build-machine-running-out-of-disk-space.html seems to indicate that may not be the case...

```
Hi, after we check and confirm, as you know, the hosted agent is redistribute one new agent to you while build running.

Though the agent has 10 GB space, but it was used by other user before it is distribute to you.

Now, the issue caused by that the space of agent is not cleared cleanly after recycling this agent.
```

Reply:
I also started this thread in the community forum about this. After a bit of back and forth I was never able to get clear answers/solutions though and we stopped seeing the issues with mine & Anthony's fixes so I stopped updating the thread. Might want to start replying again though since it's very weird that this keeps happening suddenly.